v2/internal/testcommon/samples_tester.go

```diff
@@ -228,3 +235,43 @@ func IsSampleExcluded(path string, exclusions []string) bool {
 	}
 	return false
 }
+
+func (t *SamplesTester) findAndSetARMIDReferences(obj genruntime.ARMMetaObject) (genruntime.ARMMetaObject, error) {
+	resourceRefType := reflect.TypeOf(genruntime.ResourceReference{})
+	visitor := reflecthelpers.NewReflectVisitor()
+	visitor.VisitStruct = func(this *reflecthelpers.ReflectVisitor, it reflect.Value, ctx interface{}) error {
+
+		if it.Type() == resourceRefType {
+			if it.CanInterface() {
+				reference := it.Interface().(genruntime.ResourceReference)
+				if reference.ARMID != "" {
+					armIDField := it.FieldByName("ARMID")
+					if !armIDField.CanSet() {
+						return errors.New("cannot set 'ARMID' field of 'genruntime.ResourceReference'")
+					}
+
+					//Regex to match '/00000000-0000-0000-0000-000000000000/' strings, to replace with the subscriptionID
+					subMatcher := regexp.MustCompile("\\/([0]+-?)+\\/")
+					armIDString := armIDField.String()
+					armIDString = strings.ReplaceAll(armIDString, defaultResourceGroup, t.rgName)
+					armIDString = subMatcher.ReplaceAllString(armIDString, fmt.Sprint("/", t.azureSubscription, "/"))
+
+					armIDField.SetString(armIDString)
+				}
+			} else {
+				// This should be impossible given how the visitor works
+				return errors.New("genruntime.ResourceReference field was unexpectedly nil")
```

Comment:
How does this error message follow from `!t.CanInterface()` returning false?

(As in, if I got this error and found this code by `grep`, how do I tell how to fix my problem?)

Reply:
Actually, if it's genuinely impossible, either remove the else branch or make it a panic.

Reply:
Actually, good idea to panic here. If its impossible, then something is really not going fine. Updating it to look like below.
`panic(fmt.Sprintf("genruntime.ResourceReference field was unexpectedly nil for '%s'", obj.GetObjectKind().GroupVersionKind().String()))`

Reply:
Actually, good idea to panic here. If its impossible, then something is really not going fine.